src/Enzo/enzo_EnzoConstrainedTransport.hpp

```diff
@@ -12,166 +12,304 @@ class EnzoConstrainedTransport
   /// @class    EnzoConstrainedTransport
   /// @ingroup  Enzo
   /// @brief    [\ref Enzo] Encapsulates operations of constrained transport
+  ///
+  /// Instances of this class are meant to be used as components of an MHD or
+  /// hydrodynamical integrators that are entirely responsible for all
+  /// constrained-transport related operations that would not otherwise be
+  /// performed by the solver. The distinction between whether a pointer is
+  /// NULL or points to an instance of this class should entirely dictate
+  /// whether constrained-transport related operations are used in the
+  /// integrator.
+  ///
+  /// To accomplish these goals, this object is responsible for the
+  /// allocation/deallocation of the required temporary scratch-space
+  /// and providing the actual constrained transport operations. To manage
+  /// scratch-space in a thread-safe manner (i.e. allocating/deallocating it
+  /// separately for each block) while avoiding two-step constructing and
+  /// destruction of instances of this class, the temporary scratch-space
+  /// arrays are implemented as temporary fields that are allocated/deallocated
+  /// in the class's constructor/destructor. As a consequence, instances of
+  /// this class need to be constructed and destroyed separately for each leaf
+  /// block during every cycle that the underlying MHD integrator is applied.
+  /// The instance also internally tracks the current partial timestep to be
+  /// less intrusive for the underlying MHD integrator.
+  ///
+  /// Although this design is crude, it is simple. The design could be improved
+  /// by making separating the responsiblities of the allocation/deallocation
+  /// of temporary scratch space and the implementation of the constrained
+  /// transport operation between two classes. However this refactoring has
+  /// been defered to the future when other classes may be introduced to
+  /// encapsulate other approaches to updated the magnetic field (e.g. Dedner
+  /// divergence cleaning).
```

Comment:
You're absolutely right. I've started to give this some thought and the thing that I'm really concerned about is reentrancy. For example if somebody plopped a Charm++ reduction into the middle of `EnzoMethodMHDVlct::compute` (for example, to facillitate Riemann Solver fallback), this could break Constrained Transport in unexpected ways if we aren't careful.

In reality, some comments about this in the code is probably sufficient.